### PR TITLE
Use table summary for large groups of dependencies

### PR DESCRIPTION
**Pull request by bdragon**
When a pull request is being created that updates five or more dependencies, render the updates as a table for legibility.

### CODE REVIEW

File: common/lib/dependabot/pull_request_creator/message_builder.rb [hunk 1]
```diff
@@ -414,6 +425,24 @@ def metadata_links_for_dep(dep)
...
+        [
+          table_row(row),
+          table_row(["---"] * row.count),
+        ].join("\n")
+      end
+
+      def table_row(row)
+        "| #{row.join(' | ')} |"
+      end
+
```
Comment (Nishnha) [inline common/lib/dependabot/pull_request_creator/message_builder.rb:445]: This is a really clean interface. Thanks!

File: common/lib/dependabot/pull_request_creator/message_builder.rb [hunk 2]
```diff
@@ -315,12 +315,19 @@ def transitive_removed_dependency_intro
...
-        msg = "Bumps the #{dependency_group.name} group#{pr_name_directory} with #{update_count} update"
-        msg += if update_count > 1
-                 "s: #{dependency_links[0..-2].join(', ')} and #{dependency_links[-1]}."
+        msg = "Bumps the #{dependency_group.name} group#{pr_name_directory} " \
+          "with #{update_count} update#{update_count > 1 ? 's' : ''}:"
+
+        msg += if update_count >= 5
+                header = ["Package", "Update"]
+                rows = dependencies.map { |dep| [dependency_link(dep), dependency_version_update(dep)] }
+                "\n\n#{table([header] + rows)}"
```
Comment (Nishnha) [inline common/lib/dependabot/pull_request_creator/message_builder.rb:324]: I think the indenting here is off?

File: common/lib/dependabot/pull_request_creator/message_builder.rb [hunk 3]
```diff
@@ -315,12 +315,19 @@ def transitive_removed_dependency_intro
...
+               elsif update_count > 1
+                 " #{dependency_links[0..-2].join(', ')} and #{dependency_links[-1]}."
                else
-                 ": #{dependency_links.first}."
+                 " #{dependency_links.first}."
                end
```
[the remark this answers is not in the document]
Comment (bdragon) [reply common/lib/dependabot/pull_request_creator/message_builder.rb:329]: That should be covered by a few existing tests:

* https://github.com/dependabot/dependabot-core/blob/86b7340f537ede7c8281cefc6bd828e4bed4e35e/common/spec/dependabot/pull_request_creator/message_builder_spec.rb#L1878
* https://github.com/dependabot/dependabot-core/blob/86b7340f537ede7c8281cefc6bd828e4bed4e35e/common/spec/dependabot/pull_request_creator/message_builder_spec.rb#L1951
* https://github.com/dependabot/dependabot-core/blob/86b7340f537ede7c8281cefc6bd828e4bed4e35e/common/spec/dependabot/pull_request_creator/message_builder_spec.rb#L1972

**Review by brrygrdn — APPROVED** [review]
🚀

**Pull request merged**